org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/catalog/XMLCatalogDiagnosticsParticipant.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lemminx.extensions.catalog;
+
+import java.util.List;
+
+import org.apache.xerces.xni.parser.XMLEntityResolver;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.extensions.contentmodel.participants.XMLSchemaErrorCode;
+import org.eclipse.lemminx.extensions.contentmodel.participants.diagnostics.XMLValidator;
+import org.eclipse.lemminx.extensions.contentmodel.settings.XMLValidationSettings;
+import org.eclipse.lemminx.extensions.xerces.LSPXMLEntityResolver;
+import org.eclipse.lemminx.services.extensions.diagnostics.DiagnosticsResult;
+import org.eclipse.lemminx.services.extensions.diagnostics.IDiagnosticsParticipant;
+import org.eclipse.lemminx.utils.DOMUtils;
+import org.eclipse.lemminx.utils.FilesUtils;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+/**
+ * Validate XML files with Xerces for general SYNTAX validation and XML Schema,
+ * DTD.
+ *
+ */
+public class XMLCatalogDiagnosticsParticipant implements IDiagnosticsParticipant {
+
+	private final XMLCatalogPlugin xmlCatalogPlugin;
+
+	public XMLCatalogDiagnosticsParticipant(XMLCatalogPlugin xmlCatalogPlugin) {
+		this.xmlCatalogPlugin = xmlCatalogPlugin;
+	}
+
+	@Override
+	public void doDiagnostics(DOMDocument xmlDocument, List<Diagnostic> diagnostics,
+			XMLValidationSettings validationSettings, CancelChecker monitor) {
+		if (!DOMUtils.isCatalog(xmlDocument)) {
+			return;
+		}
+
+		for (CatalogEntry catalogEntry : CatalogUtils.getCatalogEntries(xmlDocument)) {
+			String path = CatalogUtils.getResolvedLocation(xmlDocument, catalogEntry);
+			if (!FilesUtils.isValidPath(FilesUtils.getPath(path))) {
+				Range range = XMLPositionUtility.selectValueWithoutQuote(catalogEntry.getLinkRange());
+				diagnostics
+						.add(new Diagnostic(range, "File referred to by the catalog entry cannot be found.",
```

Comment:
I would add the uri in the error message because please keep in mind this error message appears in the Problem view.

To build the error message I suggest that you use MessageFormat.format